include/drivers/can.h

```diff
@@ -728,12 +730,16 @@ static inline int can_set_bitrate(const struct device *dev,
 		return -EINVAL;
 	}
 
+	timing.sjw = 0;
```

Comment:
I don't know, this seems sort of hacky way to do it. 
Only to allow 0 as a valid value for timing.sjw all `set_timing` functions get some backup storage for an old sjw value that should be properly initialized but isn't really guaranteed to be. In the mcp2515 case it will be initialized to 0 which is valid (sjw of 1), but even that made me wonder if I am missing a detail.
Wouldn't the proper change be to add a sjw parameter to can_set_bitrate? Don't want to break the interface eh?

Reply:
The sjw always gets initialized by the device-tree. That is why the approach is valid. The reason why I don't want to have a parameter is that the user would have to extract it from the device-tree or come up with a new value. Not changing it was the best solution IMO.

Reply:
Fair enough I forgot that it is always required in the devicetree. Could you instead of plain 0 add a define to the can.h header?
#define CAN_SJW_NOCHANGE 0?

I find the code really hard to read when just comparing against 0, then it at least makes sense what the intention is in the drivers

Reply:
Done

Reply:
thanks :+1: